galaxy_ng/app/api/ui/viewsets/collection.py

```diff
@@ -127,39 +129,36 @@ class CollectionVersionViewSet(api_base.GenericViewSet):
     serializer_class = serializers.CollectionVersionSerializer
 
     def list(self, request, *args, **kwargs):
-        self.paginator.init_from_request(request)
-
-        params = self.request.query_params.dict()
-        params.update({
-            'offset': self.paginator.offset,
-            'limit': self.paginator.limit,
-        })
-
-        # pulp uses ordering, but the UI is standardized around sort
-        if params.get('sort'):
-            params['ordering'] = params.get('sort')
-            del params['sort']
+        queryset = CollectionVersion.objects.all()
+        repo = self.request.query_params.dict().get('repository')
+        if repo:
+            try:
+                distro = AnsibleDistribution.objects.get(base_path=repo)
```

Comment:
The pulp_ansible collection viewsets have some mixins and whatnot for doing some of this.
https://github.com/pulp/pulp_ansible/blob/master/pulp_ansible/app/galaxy/v3/views.py#L38

Not sure if the pulp mixin approach is worthwhile, but it's an option. Might make it a little easier to bring the ui collections views and the v3 / pulp_ansible collection viewsets closer together.

Reply:
Yeah that probably is worth a try

Reply:
This worked except we don't want the 404 when the repo is not found, so I scraped the mixin